librz/util/vector.c

```diff
@@ -444,6 +444,29 @@ RZ_API RZ_BORROW void **rz_pvector_find(RZ_NONNULL const RzPVector *vec, RZ_NONN
 	return NULL;
 }
 
+/**
+ * \brief Joins 2 pvector into one (pvec2 pointer needs to be freed by the user)
+ *
+ **/
+
+RZ_API bool rz_pvector_join(RZ_NONNULL RzPVector *pvec1, RZ_NONNULL RzPVector *pvec2) {
+	rz_return_val_if_fail(pvec1 && pvec2, 0);
+
+	if (rz_pvector_empty(pvec2)) {
+		return false;
+	}
+
+	void **it;
+	rz_pvector_foreach (pvec2, it) {
+		rz_pvector_push(pvec1, *it);
```

Comment:
Can't we use `memmove` to speed up this somehow? Both are contiguous chunks of memory.